internal/guest/amdsev/report.go

```diff
@@ -0,0 +1,266 @@
+//go:build linux
+// +build linux
+
+// Package amdsev contains minimal functionality required to fetch
+// attestation reports inside an enlightened guest.
+package amdsev
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/guest/linux"
+)
+
+const (
+	MsgTypeInvalid = iota
+	MsgCPUIDRequest
+	MsgCPUIDResponse
+	MsgKeyRequest
+	MsgKeyResponse
+	MsgReportRequest
+	MsgReportResponse
+	MsgExportRequest
+	MsgExportResponse
+	MsgImportRequest
+	MsgImportResponse
+	MsgAbsorbRequest
+	MsgAbsorbResponse
+	MsgVMRKRequest
+	MsgVMRKResponse
+	MsgTypeMax
+)
+
+type SevSNPGuestRequest struct {
+	RequestMsgType  byte
+	ResponseMsgType byte
+	MsgVersion      byte
+	RequestLength   uint16
+	RequestUAddr    uint64
+	ResponseLength  uint16
+	ResponseUAddr   uint64
+	Error           uint32
+}
+
+// AMD SEV ioctl definitions
+const (
+	// SEV-SNP IOCTL type
+	sevGuestType = 'S'
```

Comment:
We should in general remove "sev" prefix from identifiers in this package, since that's part of the package name.